controllers/storagecluster/onboarding_validation_keys_generator_job.go

```diff
@@ -0,0 +1,114 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
+)
+
+const (
+	providerAPIServerImage              = "PROVIDER_API_SERVER_IMAGE"
+	onboardingSecretGeneratorImage      = "ONBOARDING_SECRET_GENERATOR_IMAGE"
+	onboardingJobName                   = "onboarding-secret-generator"
+	onboardingTicketPublicKeySecretName = "onboarding-ticket-key"
+)
+
+type onboardingValidationKeysGeneratorJob struct{}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	if instance.Spec.AllowRemoteStorageConsumers {
+		r.Log.Info("onboarding validation key generator job: Spec.AllowRemoteStorageConsumers is enabled")
+		if res, err := o.createJob(r, instance); err != nil {
+			return reconcile.Result{}, err
+		} else if !res.IsZero() {
+			return res, nil
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (o *onboardingValidationKeysGeneratorJob) ensureDeleted(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+
+	onboardingSecretGeneratorJob := getOnboardingJobObject(instance)
+	err := r.Client.Delete(context.TODO(), onboardingSecretGeneratorJob)
+
+	if err != nil && errors.IsNotFound(err) {
+		r.Log.Info("Failed to delete resource", "Kind", onboardingSecretGeneratorJob.GetObjectKind(), "Name", onboardingSecretGeneratorJob.GetName())
```

Comment:
Why do we need to log the Kind? we know this is a job by the context of the code

Reply:
Fairly standard practice across the project, makes it easier to skim the logs.

Reply:
In that case why just the Kind and not the entire GVK, why not the namespace name? 
our logs then to bloat. When I see a preference for one type of information while others are ignored it raises the question of standard.  

Add to that the this log message is at a place in the code where we know exactly the type we are referring to, it is not a generic function handling multiple types, so the log record should be much much simpler and should include the underlying error (which is the most important part of the log message)
```suggestion
		r.Log.Infof("Failed to delete secret %s: %v\m", onboardingSecretGeneratorJob.Name, err)
```

Reply:
I don't exactly remember why this is the case, but it is the project standard to use `r.Log.Info` with value-pairs of "Kind" and "Name". Nowhere else do we use `r.Log.Infof` in a controller. Without a deeper investigation as to why we're not using `Infof` to begin with, I don't think it's fair to arbitrarily start using it here.